src/nestedtext.zig

```diff
@@ -203,6 +207,9 @@ fn fromJsonInternal(allocator: *Allocator, json_value: json.Value) anyerror!Valu
                 .String => |inner| {
                     try buffer.writer().print("{s}", .{inner});
                 },
+                .NumberString => |inner| {
+                    try buffer.writer().print("{e}", .{std.fmt.fmtSliceEscapeLower(inner)});
```

Comment:
`NumberString` seems to be a workaround for not being able to support arbitrary length integers:
```zig
fn parseNumber(p: *Parser, n: std.meta.TagPayload(Token, Token.Number), input: []const u8, i: usize) !Value {
    return if (n.is_integer)
        Value{
            .Integer = std.fmt.parseInt(i64, n.slice(input, i), 10) catch |e| switch (e) {
                error.Overflow => return Value{ .NumberString = n.slice(input, i) },
                error.InvalidCharacter => |err| return err,
            },
        }
    else
        Value{ .Float = try std.fmt.parseFloat(f64, n.slice(input, i)) };
}
```

As per `jsonStringify()` in `std.json`:
```zig
.NumberString => |inner| try out_stream.writeAll(inner),
```
I think we want to just write the string rather than formatting it in any way.

Reply:
Can't really look at it until later, but that sounds reasonable to just treat it as a string.